### PR TITLE
cmake: Add COMPILE_QML and DEBUG_QML options

### PR DESCRIPTION
**Pull request by patrickelectric**
This helps to improve the QML performance with COMPILE_QML and debug QML problems with DEBUG_QML

@dagar more cmake updates

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
> DEBUG_QML

This should not be on by default. It causes random crashes in various OS when debugging. It's only usable for really specific stuff. I always have to turn it of in Qt Creator qmake builds.

**Comment by dagar** [comment]
Awesome!

**Comment by dagar** [comment]
FYI both options are off by default.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.12)
```
Comment (dagar) [inline CMakeLists.txt]: Any objection to leaving it at 3.10 for now? That's the cmake version that ships with the current Ubuntu LTS (18.04), and where a lot of developers/users live.

Comment (patrickelectric) [reply CMakeLists.txt]: Moved to 3.10

**Pull request merged**